backend/src/services/company.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Request } from 'express';
+import { Company } from '../data/entities/company';
+import { HttpCode, HttpError } from 'growup-shared';
+
+export const createCompany = async (data: Request): Promise<Company> => {
+  const { name } = data.body;
+
+  if (name) {
+    const isCompanyExist = await Company.find({ name });
+
+    if (isCompanyExist.length === 0) {
+      const company = new Company();
+      const newCompany = Object.assign(company, data.body);
+
+      await Company.save(newCompany);
+      return Promise.resolve(newCompany);
```

Comment:
Promise resolve is not needed
```suggestion
      return newCompany;
```